atlas/src/shared/components/Carousel/Carousel.tsx

```diff
@@ -1,94 +1,70 @@
-import React, { useState } from 'react'
-import { SerializedStyles } from '@emotion/core'
-import { animated, useSpring } from 'react-spring'
-import useResizeObserver from 'use-resize-observer'
-import { CarouselStyleProps, useCSS } from './Carousel.style'
-import NavButton from '../NavButton'
+import React, { useState, useEffect, useRef } from 'react'
+import { GliderProps, BreakPoint, GliderMethods } from 'react-glider'
 
-export type CarouselProps = {
-  containerCss?: SerializedStyles
-  leftControlCss?: SerializedStyles
-  rightControlCss?: SerializedStyles
-  disableControls?: boolean
-  onScroll?: (direction: 'left' | 'right') => void
-} & CarouselStyleProps
+import { Container, StyledGlider, Arrow } from './Carousel.style'
 
-const Carousel: React.FC<CarouselProps> = ({
-  children,
-  containerCss,
-  leftControlCss,
-  rightControlCss,
-  disableControls = false,
-  onScroll = () => {},
-}) => {
-  const [scroll, setScroll] = useSpring(() => ({
-    transform: `translateX(0px)`,
-  }))
-  const [carouselOffset, setCarouselOffset] = useState(0)
-  const { width: containerWidth = 0, ref: containerRef } = useResizeObserver<HTMLDivElement>()
-  const { width: childrenWidth = 0, ref: childrenContainerRef } = useResizeObserver<HTMLDivElement>()
+import 'glider-js/glider.min.css'
 
-  const styles = useCSS({})
+type CarouselProps = {
+  trackPadding?: string
+} & GliderProps
 
-  const maxScrollOffset = childrenWidth - containerWidth
+type TrackProps = {
+  className?: string
+  padding?: string
+}
+const Track: React.FC<TrackProps> = ({ className = '', ...props }) => (
+  <div className={`glider-track ${className}`} {...props} />
+)
 
-  const showLeftControl = !disableControls && carouselOffset > 0
-  const showRightControl = !disableControls && carouselOffset < maxScrollOffset
+const RightArrow = <Arrow name="chevron-right" />
+const LeftArrow = <Arrow name="chevron-left" />
 
-  const handleScroll = (direction: 'left' | 'right') => {
-    if (containerWidth == null) {
-      return
-    }
-    let scrollAmount
-    switch (direction) {
-      case 'left': {
-        // Prevent overscroll on the left
-        const newOffset = carouselOffset - containerWidth
-        scrollAmount = newOffset < 0 ? 0 : newOffset
-        onScroll('left')
-        break
-      }
-      case 'right': {
-        // Prevent overscroll on the right
-        const newOffset = carouselOffset + containerWidth
-        scrollAmount = newOffset > maxScrollOffset ? maxScrollOffset : newOffset
-        onScroll('right')
-        break
-      }
+const Carousel: React.FC<CarouselProps> = ({ children, trackPadding = '0', className, ...gliderProps }) => {
+  // Using any because the GliderMethods type only has methods and I need the full instance
+  const gliderRef = React.useRef<any>()
+  const [arrows, setArrows] = React.useState<
+    { prev: string | HTMLButtonElement; next: string | HTMLButtonElement } | undefined
```

Comment:
Looking at the code, those always will be refs, not strings, why this type?

Reply:
This was mainly because the actual correct type of `gliderjs` accepts both a string and an element, I've changed this and added a type conversion when passing them to the `Glider` component (since `react-glider` expects them both to be strings)